syntax = "proto3";

package NCloud.NFileStore.NProto;

import "cloud/storage/core/protos/request_source.proto";
import "cloud/storage/core/protos/throttler.proto";
import "cloud/storage/core/protos/trace.proto";
import "library/cpp/lwtrace/protos/lwtrace.proto";

option go_package = "github.com/ydb-platform/nbs/cloud/filestore/public/api/protos";

////////////////////////////////////////////////////////////////////////////////
// Common message headers.

message THeaders
{
    // Trace identifier for logging.
    string TraceId = 1;

    // Idempotence identifier for retries.
    string IdempotenceId = 2;

    // Client identifier for client detection.
    string ClientId = 3;

    // Session identifier.
    string SessionId = 4;

    // Request timestamp.
    uint64 Timestamp = 5;

    // Request identifier.
    uint64 RequestId = 6;

    // Request timeout (in milliseconds).
    uint32 RequestTimeout = 7;

    // Session sequence endpoint set by endpoint.
    uint64 SessionSeqNo = 8;

    message TInternal
    {
        NLWTrace.TTraceRequest Trace = 1;

        NCloud.NProto.ERequestSource RequestSource = 2;

        // IAM auth token.
        string AuthToken = 3;
    }

    // Internal header, should not be used outside.
    TInternal Internal = 9;

    // FQDN of the host, where the request originated.
    string OriginFqdn = 10;

    // Explicitly disables multitablet forwarding.
    bool DisableMultiTabletForwarding = 11;

    // Throttling already performed.
    bool ThrottlingDisabled = 12;

    // Behave as directory tablet (main tablet) - allows work delegation to
    // shards if directory handling is allowed for shards.
    bool BehaveAsDirectoryTablet = 13;
}

////////////////////////////////////////////////////////////////////////////////
// Common response headers.

message TResponseHeaders
{
    // Response traces.
    NCloud.NProto.TTraceInfo Trace = 1;

    // Response throttler info.
    NCloud.NProto.TThrottlerInfo Throttler = 2;
}
